pkg/grpc/api.go

```diff
@@ -756,29 +780,35 @@ func (a *api) GetBulkState(ctx context.Context, in *runtimev1pb.GetBulkStateRequ
 	}
 
 	start := time.Now()
-	var bulkGet bool
-	var responses []state.BulkGetResponse
 	policy := a.resiliency.ComponentOutboundPolicy(ctx, in.StoreName, resiliency.Statestore)
-	err = policy(func(ctx context.Context) (rErr error) {
-		bulkGet, responses, rErr = store.BulkGet(reqs)
-		return rErr
+	bgrAny, err := policy(func(ctx context.Context) (any, error) {
+		rBulkGet, rBulkResponse, rErr := store.BulkGet(reqs)
+		return &bulkGetRes{
+			bulkGet:   rBulkGet,
+			responses: rBulkResponse,
+		}, rErr
 	})
 	elapsed := diag.ElapsedSince(start)
 
 	diag.DefaultComponentMonitoring.StateInvoked(ctx, in.StoreName, diag.BulkGet, err == nil, elapsed)
 
+	bgr, _ := bgrAny.(*bulkGetRes)
+	if bgr == nil {
+		bgr = &bulkGetRes{}
+	}
```

Comment:
If every single retry times out, this could happen, as in that case the policy would return an error and `nil` data.

I can look into adding a unit test

Reply:
Added unit test